### ISSUE TITLE
Build failures with GHC 8.10.1

### ISSUE THREAD

**Issue by felixonmars** [sebastiaanvisser/fclabels#40]
```
[ 8 of 10] Compiling Data.Label.Base  ( src/Data/Label/Base.hs, dist/build/Data/Label/Base.dyn_o )

src/Data/Label/Base.hs:79:10: error:
    • Couldn't match expected type ‘Poly.Lens
                                      cat (Maybe a -> Maybe b) (a -> b)’
                  with actual type ‘Unit
                                      (Poly.Lens cat0 (Maybe a0 -> Maybe b0) (a0 -> b0))’
    • In the expression:
        (Unit (point
                 $ (Data.Label.Point.Point
                      ((Control.Arrow.zeroArrow Control.Arrow.||| Control.Category.id)
                         Control.Category.<<<
                           arr
                             (\ f
                                -> case f of
                                     Just a -> Right a
                                     _ -> Left ())))
                     ((Data.Label.Derive.modifier
                         ((Control.Arrow.zeroArrow Control.Arrow.||| Control.Category.id)
                            Control.Category.<<<
                              arr
                                (\ f
                                   -> case f of
                                        Just a -> Right a
                                        _ -> Left ())))
                        ((Control.Arrow.zeroArrow Control.Arrow.||| Control.Category.id)
                           Control.Category.<<<
                             arr
                               (\ (v, f)
                                  -> case f of
                                       Just _ -> Right (Just v)
                                       _ -> Left ())))) ::
           forall cat a b.
           (ArrowChoice cat, ArrowApply cat, ArrowZero cat) =>
           Unit Poly.Lens cat (Maybe a -> Maybe b) (a -> b))
      In an equation for ‘just’:
          just
            = (Unit (point
                       $ (Data.Label.Point.Point
                            ((Control.Arrow.zeroArrow Control.Arrow.||| Control.Category.id)
                               Control.Category.<<<
                                 arr
                                   (\ f
                                      -> case f of
                                           Just a -> Right a
                                           _ -> Left ())))
                           ((Data.Label.Derive.modifier
                               ((Control.Arrow.zeroArrow Control.Arrow.||| Control.Category.id)
                                  Control.Category.<<<
                                    arr
                                      (\ f
                                         -> case f of
                                              Just a -> Right a
                                              _ -> Left ())))
                              ((Control.Arrow.zeroArrow Control.Arrow.||| Control.Category.id)
                                 Control.Category.<<<
                                   arr
                                     (\ (v, f)
                                        -> case f of
                                             Just _ -> Right (Just v)
                                             _ -> Left ())))) ::
                 forall cat a b.
                 (ArrowChoice cat, ArrowApply cat, ArrowZero cat) =>
                 Unit Poly.Lens cat (Maybe a -> Maybe b) (a -> b))
    • Relevant bindings include
        just :: Poly.Lens cat (Maybe a -> Maybe b) (a -> b)
          (bound at src/Data/Label/Base.hs:79:1)
   |
79 | just = $(getLabel ''Maybe)
   |          ^^^^^^^^^^^^^^^^
```

**Comment by hesselink** [comment]
Looks to be due to a template haskell change:
> Using TupleT 1, TupE [exp], or TupP [pat] will now produce unary tuples (i.e., involving the Unit type from GHC.Tuple) instead of silently dropping the parentheses. This brings Template Haskell’s treatment of boxed tuples in line with that of unboxed tuples, as UnboxedTupleT`, ``UnboxedTupE, and UnboxedTupP also produce unary unboxed tuples (i.e., Unit#) when applied to only one argument.

Looks like we'll have to either special case the one constructor case in `getLabel`, or pattern match on `Unit` at all use sites. I'd prefer the former. PR welcome if this is urgent for you!

**Comment by hesselink** [comment]
Fixed by #41
